src/adr/0040-generalize-event-bus.md

```diff
@@ -0,0 +1,30 @@
+#40. Generalize EventBus
+
+Date: 2020-06-11
+
+## Status
+
+Proposed
+
+## Context
+
+When searching for users by quota ratio if we set the value of the parameters to 0, the search feature is supposed to return newly created users
+who have not receive any email yet at that point. However, this is not the case due to the quotas are currently being initialize only after
+an user receive the first email.
+
+In order to solve this problem, we need to have some kind of event-driven system so when an new user is created successfully, 
+their quota should also be created immediately after. 
+
+The EventBus components currently implemented in mailbox-api would be suitable for this purpose.
+
+## Decision
+
+Move the EventBus outside of mailbox-api so it can be implemented anywhere in James by adding the corresponding dependency. 
```

Comment:
so it can be used by all components

Reply:
I think we miss a term for components, that are not core components, but offer services to other core components. I propose utility component wording.

Then we just say "extract the EventBus out of mailbox-api in order to make it a utility component".

What do you think?